common/src/state/mod.rs

```diff
@@ -973,6 +987,20 @@ impl State {
             }
             message.set_reactions(reactions);
             msg.inner = message;
+            if pinned_change {
```

Comment:
I believe the logic here is backwards. if `message` is pinned then it should be added to `conv.pinned_messages` but on line 993 `.retain()` is being used to remove messages from `pinned_messages`. 


I believe you also may not need to call `.filter()` on `conv.messages` (line 999). I think it should be enough to either add or remove the message from `conv.pinned_messages`. 

in `MessageEvent::Deleted` (line 497), need to check if the message was pinned and if so, update `conv.pinned_messages`

Reply:
delete is fixued

Reply:
i read lines 1000-1001 as "if incoming message is pinned and the pin state has changed..."  But I see that was a mistake on my part due to line 999. 

I'm confused about where `pinned_messages` gets updated. I'll keep looking for that. 

Now that I think about it, that entire `if` statement from lines 1001 to 1012 could be replaced by lines 1005 to 1011. That would eliminate one possible source of confusion. 

It would probably be less confusing if line 999 was done away with. It looks like `msg` isn't assigned to anything and I think it would be more clear if `message` was used to update `conv.pinned_messages`.

Reply:
actually line 999 is not for pinned things. `update_message` was originally only used for when reactions are updated. i kinda just hijacked the function and added the pin update handling there since else i would need to copy all the things into a new function.

Reply:
true, and `msg` is a `mut` reference to `conv.messages`.

Reply:
hmm wondering then if this can be closed as resolved?

Reply:
closing it then. seems its addressed